docs/report.md

```diff
@@ -26,6 +26,24 @@ Here comes a description of our domain model.
+
+When the callback URL with the access token arrives in the Azure Tenant, it checks if the access_token have the necessary claims. Since we need the user's email, and Github doesn't provide this, our Azure Tenant needs to provide it for us. If the user exists as a user in the Azure Tenant, then the user's email is provided there, and it can be fetched from there. Otherwise, the Azure Tenant sends a dialog to the client where the user can fill in the email input. Now the Azure Tenant can create a user with sufficient claims and send the access token to the App service through the callback URL provided in the connection strings of the App Service.
+
+Using Microsoft Identity, our web application now sets properties like name, email, and isAuthenticated to the values provided by the access token. Before accessing our Chirp! web application, the login redirects to an AuthorAuthenticator. Our user is at this point stored in the Azure Tenant but may not yet be stored in our SQL Server database, which we need to ensure.
+
+![Sequence diagram of register as a User on _Chirp!_.\lable{UserRegistration}](https://raw.githubusercontent.com/ITU-BDSA23-GROUP12/Chirp/main/docs/images/SequenceOfFunctionality-RegisterUser.drawio.png)
+
+Therefore, as shown in figure \ref{AuthorAuthentication}, when landing on the AuthorAuthenticator page, a GET request is made to the PageModel, and a method on our AuthorRepository is called to find the Author, assosiated with the users email. Since the AuthorRepository has the ChirpDBContext injected as a dependency in its constructor, it now calls GetFirstORDefaultAsync, with the users email as argument, on the ChirpDBContext's Authors DbSet. EF Core now uses the connection string provided to the the ChirpDBContext from the Azure SQL Server, to make a TCP connection to the MSQL database and is now able to query it to finde the Author record with the given email and return it back to the ChirpDBContext.
```

Comment:
```suggestion
Therefore, as shown in figure \ref{AuthorAuthentication}, when landing on the AuthorAuthenticator page, a GET request is made to the PageModel, and a method on our AuthorRepository is called to find the Author, assosiated with the users email. Since the AuthorRepository has the ChirpDBContext injected as a dependency in its constructor, it now calls GetFirstORDefaultAsync, with the users email as argument, on the ChirpDBContext's Authors DbSet. EF Core now uses the connection string provided to the ChirpDBContext from the Azure SQL Server, to make a TCP connection to the MSQL database and is now able to query it to finde the Author record with the given email and return it back to the ChirpDBContext.
```